src/vehicle_selector.h

```diff
@@ -12,9 +12,9 @@ struct tripoint;
 class vehicle_cursor : public visitable<vehicle_cursor>
 {
     public:
-        vehicle_cursor( vehicle &veh, int part ) : veh( veh ), part( part ) {};
+        vehicle_cursor( vehicle &veh, size_t part ) : veh( veh ), part( part ) {};
         vehicle &veh;
-        int part;
+        size_t part;
```

Comment:
I haven't checked relevant code, but changing signed integer to unsigned could break many things.

Reply:
I'm changing it to be `ptrdiff_t` though I think `size_t` would be fine.

The only reason these warnings fire is because they're triggered in the standard library header files due to templates delaying the point where the conversion actually happens and MSVC's headers are forced to warning level 3.